hydrolib/core/io/crosssection/models.py

```diff
@@ -639,14 +640,52 @@ class CrossSection(INIBasedModel):
 
     Attributes:
         id (str): Unique cross-section location id.
-        branchid (str): (optional) Branch on which the cross section is located.
+        branchid (str, optional): Branch on which the cross section is located.
+        chainage: (str, optional): Chainage on the branch (m).
+        x: (str, optional): x-coordinate of the location of the cross section.
+        y: (str, optional): y-coordinate of the location of the cross section.
+        shift (float, optional): Vertical shift of the cross section definition [m]. Defined positive upwards.
+        definitionid (str): Id of cross section definition.
     """
 
-    # TODO: complete support for all crosssection fields: #172.
+    class Comments(INIBasedModel.Comments):
+        id: Optional[str] = "Unique cross-section location id."
+        branchid: Optional[str] = Field(
+            "Branch on which the cross section is located.", alias="branchId"
+        )
+        chainage: Optional[str] = "Chainage on the branch (m)."
+
+        x: Optional[str] = Field(
+            "x-coordinate of the location of the cross section.",
+        )
+        y: Optional[str] = Field(
+            "y-coordinate of the location of the cross section.",
+        )
+        shift: Optional[str] = Field(
+            "Vertical shift of the cross section definition [m]. Defined positive upwards.",
+        )
+        definitionid: Optional[str] = Field(
+            "Id of cross section definition.", alias="definitionId"
+        )
+
+    comments: Comments = Comments()
 
     _header: Literal["CrossSection"] = "CrossSection"
     id: str = Field(alias="id")
-    branchid: str = Field(alias="branchId")
+
+    branchid: Optional[str] = Field(None, alias="branchId")
+    chainage: Optional[float] = Field(None)
+
+    x: Optional[float] = Field(None)
+    y: Optional[float] = Field(None)
+
+    branchid: Optional[str] = Field(alias="branchId")
+    shift: Optional[float] = Field(None)
```

Comment:
Default should be 0.0 as described in the manual

Reply:
I've fixed this. Indeed, consistent with all other *Models. Later we may want to reconsider: should defaults of the kernel be reflected in our Python classes? The alternative is to have None for most defaults, such that empty key-values are written in the produced input files, and leave it to the kernel. Food for thought later.